### ISSUE TITLE
Export of strings to l10n repository is broken for Focus

### ISSUE THREAD

**Issue by flodolo** [mozilla-mobile/firefox-ios#19302]
This worked on March 12, but broke today (March 19)

Locally I get a lot of errors in `export-strings.log` (only keeping a couple, as they repeat for different locales)

```
2024-03-19 18:59:50.361 xcodebuild[89267:10453809] Writing error result bundle to /var/folders/7k/613lfl692dq22n6q952h_syw0000gp/T/ResultBundle_2024-19-03_18-59-0050.xcresult
xcodebuild: error: Unable to build project for localization string extraction
	Reason: See the build logs for failure description.
LocalizationTools/ExportTask.swift:44: Fatal error: 'try!' expression unexpectedly raised an error: Error Domain=NSCocoaErrorDomain Code=260 "The file “he.xliff” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/tmp/ios-localization-89240/he.xcloc/Localized Contents/he.xliff, NSUnderlyingError=0x6000023fc930 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}

(...)

LocalizationTools/ExportTask.swift:44: Fatal error: 'try!' expression unexpectedly raised an error: Error Domain=NSCocoaErrorDomain Code=260 "The file “hy-AM.xliff” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/tmp/ios-localization-89240/hy-AM.xcloc/Localized Contents/hy-AM.xliff, NSUnderlyingError=0x6000023f86c0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
./focus-ios/focus-ios-tests/tools/export-strings.sh: line 36: 89240 Trace/BPT trap: 5       swift run LocalizationTools --export --project-path "$PWD/../../../Blockzilla.xcodeproj" --l10n-project-path "$PWD/../../../focusios-l10n"
```

CC isabelrios

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-8705)

**Comment by flodolo** [comment]
A couple more notes. 

There's another error that might be causing this problem

```
CompileSwift normal arm64 (in target 'WidgetsExtension' from project 'Blockzilla')
    firefox-ios/focus-ios/Widgets/Widgets.swift:8:8: error: no such module 'Widget'
import Widget
       ^
```

As for the [Bash script](https://github.com/mozilla-mobile/firefox-ios/blob/main/focus-ios/focus-ios-tests/tools/export-strings.sh). We could also only export the en-US locale instead of all locales

```bash
(cd focus-ios-tests/tools/Localizations && swift run LocalizationTools \
  --export \
  --project-path "$PWD/../../../Blockzilla.xcodeproj" \
  --l10n-project-path "$PWD/../../../focusios-l10n" \
  --locale en-US) > export-strings.log 2>&1
```

**Comment by isabelrios** [comment]
I have found when this started to happen... 
This is the PR where the the failure starts: https://github.com/mozilla-mobile/firefox-ios/commit/0194fca784ee190138615652439d9fb4ef9cee4e
It is a rust component upgrade... not sure how it is related to the failure... maybe due to the widget kit dependency?

**Comment by isabelrios** [comment]
Building via xcode works, at least for debug... the logs show that this might affect FocusRelease...
```
/Blockzilla.build/FocusRelease-iphoneos/WidgetsExtension.build/Objects-normal/arm64/AppConfig.o -index-unit-output-path /Blockzilla.build/FocusRelease-iphoneos/WidgetsExtension.build/Objects-normal/arm64/AppInfo.o -index-unit-output-path /Blockzilla.build/FocusRelease-iphoneos/WidgetsExtension.build/Objects-normal/arm64/GeneratedAssetSymbols.o
/Users/isabelrios/git/firefox-ios-no-fork/firefox-ios/focus-ios/Widgets/Widgets.swift:8:8: error: no such module 'Widget'
import Widget
```

**Comment by flodolo** [comment]
Not sure if I'm doing it wrong, but I tried to manually revert those changes and it still fails for me locally.

**Comment by isabelrios** [comment]
> Not sure if I'm doing it wrong, but I tried to manually revert those changes and it still fails for me locally.

It worked for me, reverted the focus files changes and I got the export locally working

**Comment by isabelrios** [comment]
There might be something in the cache and build that is affecting how this work locally.. still investigating, I got the failure in previous commits also

**Comment by flodolo** [comment]
Note: no need to run the script or even LocalizationTools, this should be enough to get the same failure `xcodebuild -exportLocalizations -project focus-ios/Blockzilla.xcodeproj`

**Comment by isabelrios** [comment]
There is another PR that looks suspicious to me.. https://github.com/mozilla-mobile/firefox-ios/commit/bccf2cf9382dde2b1795f4f3a0ea3ac1b2972faf cleaning the environment and building there I can't get the export working...

**Comment by isabelrios** [comment]
To add more findings... in latest main, running the export from xcode ui, the process finishes successfully, no error shown
<img width="197" alt="Screenshot 2024-03-20 at 16 51 18" src="https://github.com/mozilla-mobile/firefox-ios/assets/1897507/3da4d9c5-c7a6-439f-b9d7-e492b5b858c6">

A folder is created with the different files
<img width="1282" alt="Screenshot 2024-03-20 at 16 53 04" src="https://github.com/mozilla-mobile/firefox-ios/assets/1897507/be8c98b1-f955-40cf-b51f-67dda784a839">

So, there is a difference when running it via the command line, something when building that is failing to build all the dependencies?

**Comment by flodolo** [comment]
Not sure what I'm doing wrong, but I'm unable to build Focus even in XCode: it builds, but it fails to install on the simulator. On the other hand, I can build and run Firefox without issues. 

This is the full error, mentioning Rust Components

```
App installation failed: Installazione di Firefox Focus non riuscita
Domain: IXUserPresentableErrorDomain
Code: 1
Failure Reason: Riprova più tardi.
Recovery Suggestion: Failed to load Info.plist from bundle at path /Users/flodolo/Library/Developer/CoreSimulator/Devices/BCA20EBF-1E2C-47CD-971D-F0182EB8CE9E/data/Library/Caches/com.apple.mobile.installd.staging/temp.bVVy5L/extracted/Firefox Focus.app/Frameworks/MozillaRustComponents.framework; Extra info about "/Users/flodolo/Library/Developer/CoreSimulator/Devices/BCA20EBF-1E2C-47CD-971D-F0182EB8CE9E/data/Library/Caches/com.apple.mobile.installd.staging/temp.bVVy5L/extracted/Firefox Focus.app/Frameworks/MozillaRustComponents.framework/Info.plist": Couldn't stat /Users/flodolo/Library/Developer/CoreSimulator/Devices/BCA20EBF-1E2C-47CD-971D-F0182EB8CE9E/data/Library/Caches/com.apple.mobile.installd.staging/temp.bVVy5L/extracted/Firefox Focus.app/Frameworks/MozillaRustComponents.framework/Info.plist: No such file or directory
User Info: {
    DVTErrorCreationDateKey = "2024-03-21 07:58:00 +0000";
    IDERunOperationFailingWorker = IDELaunchiPhoneSimulatorLauncher;
    SimCallingSelector = "installApplication:withOptions:error:";
}
--
App installation failed: Installazione di Firefox Focus non riuscita
Domain: IXUserPresentableErrorDomain
Code: 1
Failure Reason: Riprova più tardi.
Recovery Suggestion: Failed to load Info.plist from bundle at path /Users/flodolo/Library/Developer/CoreSimulator/Devices/BCA20EBF-1E2C-47CD-971D-F0182EB8CE9E/data/Library/Caches/com.apple.mobile.installd.staging/temp.bVVy5L/extracted/Firefox Focus.app/Frameworks/MozillaRustComponents.framework; Extra info about "/Users/flodolo/Library/Developer/CoreSimulator/Devices/BCA20EBF-1E2C-47CD-971D-F0182EB8CE9E/data/Library/Caches/com.apple.mobile.installd.staging/temp.bVVy5L/extracted/Firefox Focus.app/Frameworks/MozillaRustComponents.framework/Info.plist": Couldn't stat /Users/flodolo/Library/Developer/CoreSimulator/Devices/BCA20EBF-1E2C-47CD-971D-F0182EB8CE9E/data/Library/Caches/com.apple.mobile.installd.staging/temp.bVVy5L/extracted/Firefox Focus.app/Frameworks/MozillaRustComponents.framework/Info.plist: No such file or directory
User Info: {
    IDERunOperationFailingWorker = IDELaunchiPhoneSimulatorLauncher;
    SimCallingSelector = "installApplication:withOptions:error:";
}
--
Installazione di Firefox Focus non riuscita
Domain: IXUserPresentableErrorDomain
Code: 1
Failure Reason: Riprova più tardi.
Recovery Suggestion: Failed to load Info.plist from bundle at path /Users/flodolo/Library/Developer/CoreSimulator/Devices/BCA20EBF-1E2C-47CD-971D-F0182EB8CE9E/data/Library/Caches/com.apple.mobile.installd.staging/temp.bVVy5L/extracted/Firefox Focus.app/Frameworks/MozillaRustComponents.framework; Extra info about "/Users/flodolo/Library/Developer/CoreSimulator/Devices/BCA20EBF-1E2C-47CD-971D-F0182EB8CE9E/data/Library/Caches/com.apple.mobile.installd.staging/temp.bVVy5L/extracted/Firefox Focus.app/Frameworks/MozillaRustComponents.framework/Info.plist": Couldn't stat /Users/flodolo/Library/Developer/CoreSimulator/Devices/BCA20EBF-1E2C-47CD-971D-F0182EB8CE9E/data/Library/Caches/com.apple.mobile.installd.staging/temp.bVVy5L/extracted/Firefox Focus.app/Frameworks/MozillaRustComponents.framework/Info.plist: No such file or directory
--
Failed to load Info.plist from bundle at path /Users/flodolo/Library/Developer/CoreSimulator/Devices/BCA20EBF-1E2C-47CD-971D-F0182EB8CE9E/data/Library/Caches/com.apple.mobile.installd.staging/temp.bVVy5L/extracted/Firefox Focus.app/Frameworks/MozillaRustComponents.framework; Extra info about "/Users/flodolo/Library/Developer/CoreSimulator/Devices/BCA20EBF-1E2C-47CD-971D-F0182EB8CE9E/data/Library/Caches/com.apple.mobile.installd.staging/temp.bVVy5L/extracted/Firefox Focus.app/Frameworks/MozillaRustComponents.framework/Info.plist": Couldn't stat /Users/flodolo/Library/Developer/CoreSimulator/Devices/BCA20EBF-1E2C-47CD-971D-F0182EB8CE9E/data/Library/Caches/com.apple.mobile.installd.staging/temp.bVVy5L/extracted/Firefox Focus.app/Frameworks/MozillaRustComponents.framework/Info.plist: No such file or directory
Domain: MIInstallerErrorDomain
Code: 35
User Info: {
    FunctionName = "-[MIBundle _validateWithError:]";
    LegacyErrorString = PackageInspectionFailed;
    SourceFileLine = 64;
}
--

Event Metadata: com.apple.dt.IDERunOperationWorkerFinished : {
    "device_model" = "iPhone16,1";
    "device_osBuild" = "17.4 (21E213)";
    "device_platform" = "com.apple.platform.iphonesimulator";
    "dvt_coredevice_version" = "355.24";
    "dvt_mobiledevice_version" = "1643.100.58";
    "launchSession_schemeCommand" = Run;
    "launchSession_state" = 1;
    "launchSession_targetArch" = arm64;
    "operation_duration_ms" = 4702;
    "operation_errorCode" = 1;
    "operation_errorDomain" = IXUserPresentableErrorDomain;
    "operation_errorWorker" = IDELaunchiPhoneSimulatorLauncher;
    "operation_name" = IDERunOperationWorkerGroup;
    "param_debugger_attachToExtensions" = 0;
    "param_debugger_attachToXPC" = 1;
    "param_debugger_type" = 3;
    "param_destination_isProxy" = 0;
    "param_destination_platform" = "com.apple.platform.iphonesimulator";
    "param_diag_MainThreadChecker_stopOnIssue" = 0;
    "param_diag_MallocStackLogging_enableDuringAttach" = 0;
    "param_diag_MallocStackLogging_enableForXPC" = 1;
    "param_diag_allowLocationSimulation" = 1;
    "param_diag_checker_tpc_enable" = 1;
    "param_diag_gpu_frameCapture_enable" = 0;
    "param_diag_gpu_shaderValidation_enable" = 0;
    "param_diag_gpu_validation_enable" = 0;
    "param_diag_memoryGraphOnResourceException" = 0;
    "param_diag_queueDebugging_enable" = 1;
    "param_diag_runtimeProfile_generate" = 0;
    "param_diag_sanitizer_asan_enable" = 0;
    "param_diag_sanitizer_tsan_enable" = 0;
    "param_diag_sanitizer_tsan_stopOnIssue" = 0;
    "param_diag_sanitizer_ubsan_stopOnIssue" = 0;
    "param_diag_showNonLocalizedStrings" = 0;
    "param_diag_viewDebugging_enabled" = 1;
    "param_diag_viewDebugging_insertDylibOnLaunch" = 1;
    "param_install_style" = 0;
    "param_launcher_UID" = 2;
    "param_launcher_allowDeviceSensorReplayData" = 0;
    "param_launcher_kind" = 0;
    "param_launcher_style" = 0;
    "param_launcher_substyle" = 0;
    "param_runnable_appExtensionHostRunMode" = 0;
    "param_runnable_productType" = "com.apple.product-type.application";
    "param_structuredConsoleMode" = 1;
    "param_testing_launchedForTesting" = 0;
    "param_testing_suppressSimulatorApp" = 0;
    "param_testing_usingCLI" = 0;
    "sdk_canonicalName" = "iphonesimulator17.4";
    "sdk_osVersion" = "17.4";
    "sdk_variant" = iphonesimulator;
}
--


System Information

macOS Version 14.4 (Build 23E214)
Xcode 15.3 (22618) (Build 15E204a)
Timestamp: 2024-03-21T08:58:00+01:00
```

Isabel also found a work-around to get the extraction to complete, and that's adding `LOCALIZATION_EXPORT_SUPPORTED = NO;` to the `release` target. The problem is that this doesn't export an entire file (`Widgets/en.lproj/Localizable.strings`) with 4 localizable strings in it.

**Comment by flodolo** [comment]
Oh, that's https://github.com/mozilla-mobile/firefox-ios/issues/19190

**Comment by isabelrios** [comment]
The debug target works, is not disabled for all targets, only that one, so the widgets file should be available ... not sure if we need it also in release though

**Comment by flodolo** [comment]
Error after #19190 changed to

```
error: no such module 'Glean'
import Glean
       ^
```

**Comment by isabelrios** [comment]
I see the same :(

**Comment by data-sync-user** [comment]
➤ Alexandru Farcasanu commented:

Norberto Andres Furlan 

May you add please a fix version for this? Thank you.

**Comment by dicarobinho** [comment]
@flodolo may you give it another try, now, with the fix mentioned above? Thank you.

@isabelrios yes, it was related to the _RustMozillaAppServices_ and should work after the PR from above (when doing it from Xcode)

**Comment by data-sync-user** [comment]
➤ Norberto Andres Furlan commented:

Alexandru Farcasanu Isabel Rios Escobar It should be 125. Need to create the backport to 125 I believe.

**Comment by data-sync-user** [comment]
➤ Andrei Bodea commented:

Hello, I see this with QA Needed, is there anything that can be checked here?

**Comment by data-sync-user** [comment]
➤ Alexandru Farcasanu commented:

Andrei Bodea I think, no. The things should work fine with this exeception: https://github.com/mozilla-mobile/firefox-ios/issues/19507 ( https://github.com/mozilla-mobile/firefox-ios/issues/19507|smart-link )

**Comment by data-sync-user** [comment]
➤ Andrei Bodea commented:

Thank you, if that is the case can you please close this issue and set the status accordingly? 
I believe Done is in this case.

**Comment by data-sync-user** [comment]
➤ Alexandru Farcasanu commented:

Sure. Sorry for moving it to QA Needed. 
Norberto Andres Furlan may we close this but of course I am still looking into the exception related to this one.

https://github.com/mozilla-mobile/firefox-ios/issues/19507 ( https://github.com/mozilla-mobile/firefox-ios/issues/19507|smart-link )

**Comment by isabelrios** [comment]
Agree, Let's close this one and continue the conversation in that one
